ruby/generate-short-url/index.rb

```diff
@@ -0,0 +1,71 @@
+require 'json'
+require 'uri'
+require 'net/http'
+
+VALID_PROVIDERS = %w[bitly tinyurl].freeze
+API_URLS = {
+  'bitly' => 'https://api-ssl.bitly.com/v4/shorten',
+  'tinyurl' => 'https://api.tinyurl.com/create'
+}.freeze
+
+def main(req, res)
+  payload = req.payload
+  env = req.env
+  # Input validation
+  validate_params(payload)
+  # Make sure we have envirnment variables required to execute
+  validate_environment_variables(env)
+  # Build request params
+  params = make_request_params(payload)
+  token = get_provider_token(payload['provider'], env)
+  uri = URI(params['url'])
+  # Make request to generate short url
+  make_request(uri, params, token, payload, res)
+end
+
+def make_request_params(payload)
+  provider = payload['provider']
+
+  return { 'url' => API_URLS[provider], 'body' => { 'long_url' => payload['url'] } } if provider == 'bitly'
+  return { 'url' => API_URLS[provider], 'body' => { 'url' => payload['url'] } } if provider == 'tinyurl'
+end
+
+def parse_response(http_response, provider)
+  body = JSON.parse(http_response.body)
+
+  return { 'success' => false, 'errors' => body['errors'] } if http_response.code != '200'
```

Comment:
Is this valid for all bitly errors? I'm sure an error can occur not related to a field, right?

Reply:
Yes, pretty much. I've tried making some other errors like trying invalid URLs, but their API is pretty crappy in that sense hahaha Even invalid URLs turn out to be valid.

Reply:
please return a 'message' as described in the issue. For example, this isn't helpful:

```shell
$ curl -H "X-Internal-Challenge: secret-key" \
>     -H "Content-Type: application/json" \
>     -X POST http://localhost:3000/ \
>     -d '{
>         "variables": {
>             "BITLY_ACCESS_TOKEN": "BITLY_ACCESS_TOKEN",
>             "TINYURL_ACCESS_TOKEN": "TINYURL_ACCESS_TOKEN"
>         },
>         "payload": {
>             "url": "https://google.com",
>             "provider": "bitly"
>         }
> }' 
{"response":{"success":false,"errors":null},"stdout":""}
```